### ISSUE TITLE
Allow closure does not get called when Route Model Binding is enabled

### ISSUE THREAD

**Issue by TUNER88** [efficiently/authority-controller#21]
As soon I start using [Route Model Binding](http://laravel.com/docs/5.0/routing#route-model-binding) allow closure does not get called and user get access to the resource. If I disable Route Model Binding everything works fine.

```php
// app/Providers/RouteServiceProvider.php
public function boot(Router $router)
{
    parent::boot($router);

    $router->model('users', 'App\User');
}
```

```php
if ($user->hasRole('admin')) {
    // works fine 
    // $authority->allow('manage', 'App\User'); 

    // closure does not get called
    $authority->allow('manage', 'App\User', function ($self, $user) {
        return false;
    });

} else {
    $authority->allow('read', 'all');
}
```

Any ideas how to fix it?
I'm using Laravel 5.

**Comment by tortuetorche** [comment]
Hi @TUNER88,

I don't used Route Model Binding, but is your code can you check if 
`$authority->deny('manage', 'App\User');` works as expected ?
Because, I think, but I'm not sure, that the Route Model Binding skip all authorizations rules of the Authority-Controller package.

Cheers,
Tortue Torche

**Comment by TUNER88** [comment]
@tortuetorche 
`$authority->deny('manage', 'App\User');` works as expected.

After hours of debugging, if found that the problem is in $this->loadResource() method.

If i **disable** Route Model Binding and dump `$this->params` I get following output:
```
[▼
  "id" => "1"
  "action" => "show"
  "controller" => "users"
]
```
`$this->loadedInstance()`returns `true` and the resource could be loaded in `$this->loadResourceInstance()`


If I **enable** Route Model Bindling, `$this->params` looks like this:
```
[▼
  "users" => User {#198 ▶}
  "action" => "show"
  "controller" => "users"
]
```

The User ResourceInstance is already injected, id is no more available, any idea how to handle this?
Maybe pass something to `$this->loadAndAuthorizeResource();`?

Thx for your support

**Comment by tortuetorche** [comment]
You're debugging informations are really useful.
I think I need to refactor the [Parameter](https://github.com/efficiently/authority-controller/blob/2.0/src/Efficiently/AuthorityController/Parameters.php) to handle params who contains objects like I do for string or integer params.

**Comment by tortuetorche** [comment]
And I think this issue also occurs with Laravel 4.

**Comment by TUNER88** [comment]
@tortuetorche any idea how I can temporary fix in my project, until you update your package?
Its really nice package, i would like continue using it, but disabling Model Binding in whole project is impossible at this moment.

**Comment by tortuetorche** [comment]
Of course, You can edit the [vendor/efficiently/authority-controller/src/Efficiently/AuthorityController/Parameters.php](https://github.com/efficiently/authority-controller/blob/c858c53704171adbad1555b9d667094a52da55d6/src/Efficiently/AuthorityController/Parameters.php#L46) file at line 46 and add the code below:
```php
if (is_a($lastRouteParamKey, '\Eloquent')) {
    $lastRouteParamKey = $lastRouteParamKey->getKey();
}
```
**Warning:** Untested code.

**Comment by tortuetorche** [comment]
I'll try to fix this issue and add tests in a couple of days.

**Comment by TUNER88** [comment]
@tortuetorche thanks for your advice!
Unfortunately I had no success with provided code.

I have forked your repository and patched the code, works fine with one model binding.
If you are interested, you can look at [my commit](https://github.com/TUNER88/authority-controller/commit/def6eb606de1bb98e412a56506359bde1a2ea280).

Now I can pass model key in the constructor:
`$this->loadAndAuthorizeResource(['modelBindingKey' => 'users']);`

Am looking forward to new version.

**Comment by tortuetorche** [comment]
Many thanks @TUNER88, I don't know yet if I'm going to use your code or find another solution but you have done a good job :+1:

**Comment by TUNER88** [comment]
@tortuetorche this is not the best solution, that's why does not have submitted a merge request.
Multiple model bindings are not supported by current solution, its not so good for nested resources.

**Comment by tortuetorche** [comment]
You're right

**Comment by TUNER88** [comment]
@tortuetorche Any progress?

**Comment by tortuetorche** [comment]
Hi @TUNER88,

> Any progress?

Sorry for the delay.
Can you update this package in your application and test again your routes?

Cheers,
Tortue Torche

**Comment by tortuetorche** [comment]
Hi @TUNER88,

I have backported the fix for `authority-controller` 1.2
If you have still this bug, feel free to reopen this issue.

Cheers,
Tortue Torche
